### ISSUE TITLE
Lints incorrectly treating `-`-prefixed throwing tests as ordinary objects

### ISSUE THREAD

**Issue by yegor256** [objectionary/lints#1129]
The EO parser marks a truthy test attribute (`++> name`) by prefixing its
`@name` with `+`, and a throwing test attribute (`--> name`) by prefixing it
with `-` (see `Suffix.attribute` in `eo-parser`). Three lints know about the
first prefix only, so every throwing test is linted as if it were an ordinary
object: `names/non-kebab-name.xsl:14`, `misc/sparse-decoration.xsl:17` and
`tests/wrong-test-order.xsl:14` all filter on `starts-with(@name, '+')`.

On `eo-runtime` of `objectionary/eo` this produces 236 bogus warnings out of
338: 98 `non-kebab-name` ("Formation name
`-on-bytes-of-wrong-size-as-i16` must be in kebab-case" — the leading hyphen
is the marker, not part of the name), 95 `sparse-decoration` (a throwing test
such as `01-01-01.as-i16 --> on-bytes-of-wrong-size-as-i16` is a single-φ
formation by construction, exactly the case `starts-with(@name, '+')` already
excuses for truthy tests), and 43 `wrong-test-order` (the last `++>` test in a
file is reported as misplaced because the `-->` tests that follow it are not
counted as tests).

The smallest fix is to widen the three predicates from
`starts-with(@name, '+')` to also accept `-`, ideally through one shared
`eo:test-name()` function in `org/eolang/funcs/`, since
`misc/incorrect-test-object-name.xsl:14`,
`names/unsorted-named-attributes.xsl:14`,
`tests/unit-test-without-phi.xsl:14` and `tests/unit-test-missing.xsl:14`
carry the same assumption and will need it too.

**Comment by yegor256** [comment]
@volodya-lombrozo could you take a look when you have a moment? Happy to clarify anything in the report that is unclear.

**Comment by yegor256** [comment]
This is still reproducible with `org.eolang:lints:0.4.0`, so I believe the issue was
closed prematurely — the fix lives in #1136, which is still open. The three predicates
in the released jar are unchanged:

```
$ unzip -p ~/.m2/repository/org/eolang/lints/0.4.0/lints-0.4.0.jar \
    org/eolang/lints/misc/sparse-decoration.xsl | grep -o "starts-with(@name, '+')"
starts-with(@name, '+')
```

Running `mvn compile` over `eo-runtime` of `objectionary/eo` (after renaming every
truthy test to the `can-`/`cannot-`/`accepts-`/`rejects-`/`stops-on-` convention that
`bad-test-name` now requires, which brought `bad-test-name` down from 1320 to 0) leaves
exactly 253 warnings, and every single one of them is a throwing test being linted as
ordinary code:

| lint | count | cause |
| --- | --- | --- |
| `non-kebab-name` | 106 | `-stops-on-bytes-of-wrong-size` "must be in kebab-case" |
| `sparse-decoration` | 102 | a `-->` test is a single-φ formation by construction |
| `wrong-test-order` | 44 | the last `++>` test is reported as misplaced because the `-->` tests after it are not counted as tests |
| `unit-test-missing` | 1 | `directory.open` has only `-->` tests |

I verified each of the 253 individually: the 209 `non-kebab-name` /
`sparse-decoration` warnings all point at a source line holding a `-->` declaration,
and the 44 `wrong-test-order` warnings all point at the last `++>` test of a file whose
next sibling is a `-->` test.

One more prefix worth folding into #1136: `tests/bad-test-name.xsl:14` also filters on
`starts-with(@name, '+')`, so a throwing test may be named anything at all without
being flagged, while its truthy twin must carry one of the five prefixes.

**Comment by 0crat** [comment]
@yegor256 Thanks for closing this issue! You've earned +12 points for this: +8 as a basis; +4 for resolving it in less than 12 hours. Your running score is [+2721](https://www.eolang.org/zerocracy/objectionary-vitals.html); don't forget to check your [Zerocracy account](https://www.zerocracy.com/) too).

**Comment by yegor256** [comment]
@rultor release, tag is `0.4.1`
